adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -304,7 +304,7 @@ func (h *hnsw) reassignNeighbor(neighbor uint64, deleteList helpers.AllowList, b
 		if h.isOnlyNode(&vertex{id: neighbor}, deleteList) {
 			neighborNode.Lock()
 			// delete all existing connections before re-assigning
-			neighborNode.connections = map[int][]uint64{}
+			neighborNode.connections = make([][]uint64, neighborNode.level+1)
```

Comment:
test?

Reply:
I agree that a test would be valuable here, but I think this is slightly beyond the scope of this change. Replacing an uncovered line with another uncovered line isn't a good excuse, but improving test coverage for the delete process is something that I believe can be done better in a separate PR.

Reply:
oki, do you want to make a ticket for that? Or will it come up naturally?

Reply:
I would like to loop in @aliszka here, as I believe he currently has the best overview of test coverage etc around the HNSW delete process. Either an explicit ticket or "do nothing" is fine with me.

Reply:
alright, let's let @aliszka decide then